src/dsp_tools/commands/xml_validate/models/data_deserialised.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Protocol
+
+
+@dataclass
+class DataDeserialised:
+    shortcode: str
+    default_onto: str
+    resources: list[ResourceData]
+
+
+@dataclass
+class ResourceData:
+    res_id: str
+    res_class: str
+    label: str
+    values: list[ValueData]
+
+
+@dataclass
+class ValueData(Protocol):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class BooleanValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class ColorValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class DateValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class DecimalValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class GeomValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class GeonameValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class IntValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class ListValueData(ValueData):
+    prop_name: str
+    prop_value: str
+    list_name: str
+
+
+@dataclass
+class SimpleTextData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class RichtextData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class TimeValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class UriValueData(ValueData):
+    prop_name: str
+    prop_value: str
+
+
+@dataclass
+class LinkValueData(ValueData):
+    prop_name: str
+    prop_value: str
```

Comment:
*question*: Should `prop_values` from above be `str`?

To me it seems odd that e.g. the `IntValueData` does not contain an `int` but a `str` instead. Maybe there is a reason I do not understand.

Reply:
Yes there is a reason. We expect to have invalid data, eg. a string with characters where an int should be. If I construct a literal with `rdflib` the 'correct' way:
`Literal("value", datatype=XSD.integer)`
Then an error is raised from `rdflib`: `ValueError: invalid literal for int() with base 10: 'value'`
I don't want to find these errors during the construction of the graph but through SHACL.
Another way to solve this would be to construct the graph this way:

```
try:
    Literal("value", datatype=XSD.integer)
except ValueError:
    Literal("value", datatype=XSD.string)
```

Then we could specify that it must be an `xsd:integer` in the SHACL. If the SHACL were exposed through the API then I would agree with you that this problem is best solved in DSP-TOOLS. This is not the case currently so I chose to do it like this. But I am happy to change it. However in any case the typing here would stay the same, by default if you get the content of XML it comes back as a string or `None`.

Reply:
I appreciate you explanation and it makes sense to me.

Reply:
I have thought about it and now I think I prefer your suggestion. Because we actually violate `knora-api` with this: https://github.com/dasch-swiss/dsp-tools/blob/6712bd9fd3935e3c044da55403211039c1c5c550/src/dsp_tools/resources/xml_validate/knora-api.ttl#L5043

Because we expect a string in the SHACL: https://github.com/dasch-swiss/dsp-tools/blob/6712bd9fd3935e3c044da55403211039c1c5c550/src/dsp_tools/resources/xml_validate/knora-api-shapes.ttl#L176

If we solve it with `try:` and `except` we will still get the information in the SHACL graph... I just have to verify that I get the correct errors with `rdflib`. I will look into it during the other PRs.